File: Server/FIHS/Models/Fertilizer/Fertilizer.cs
using System.ComponentModel.DataAnnotations;

namespace FIHS.Models.Fertilizer
{
    public class Fertilizer
    {
        public int Id {  get; set; }
        [Required]
        public string Name { get; set; }
        public string? ImageURL { get; set; }
        public double price { get; set; }
        [MaxLength(500)]
        public string? UsageInstructions { get; set; }
        public string? Manufactuer { get; set; }
        public string? NutrientContent { get; set; }
    }
}

File: Client/FIHS/models/Fertilizer.ts
import * as yup from 'yup'

export const FertilizerSchema = yup.object({
    name: yup.string().required('برجاء إدخال الاسم'),
    image: yup.mixed().required('برجاء إرفاق صورة'),
    manufactuer: yup.string().required('برجاء إدخال المصنع'),
    usageInstructions: yup.string().required('برجاء إدخال طرق الاستعمال')
})
export type Fertilizer = yup.InferType<typeof FertilizerSchema>